portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,10 +32,11 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  private Query query = new Query("Comment");
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
```

Comment:
This should be 
```java
 PreparedQuery results = datastore.prepare(new Query("Comment"));
```

Reply:
But then we're back to initializing query each time

Reply:
My other comment was in reference to DatastoreService, a stub which is probably "expensive" to initialize. Query is probably just a wrapper around a string and parameters, and is almost certainly "cheap" to initialize.